tests/test_bs.py

```diff
@@ -339,13 +318,9 @@ def wait_genesis():
 
 
 def api_call(client_ip, data, api, namespace):
-    p = subprocess.Popen(['./kubectl-cmd.sh', '%s' % client_ip, "%s" % data, api, namespace], stdin=subprocess.PIPE,
-                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    (out, err) = p.communicate()
-    if p.returncode != 0:
-        raise Exception('An Error raised on api_call')
-    return out
-
+    # todo: this won't work with long payloads - ( `Argument list too long` ). try port-forward ?
+    res = stream(client.CoreV1Api().connect_post_namespaced_pod_exec, name="curl", namespace=namespace, command=["curl", "-s", "--request",  "POST", "--data", data, "http://" + client_ip + ":9090/" + api], stderr=True, stdin=False, stdout=True, tty=False)
+    return res
```

Comment:
Good idea

Reply:
it works every time yay! :). we still need a better solution for longer payload.